jwst/master_background/nirspec_corrections.py

```diff
@@ -0,0 +1,26 @@
+import logging
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def correct_nrs_ifu_bkg(input):
+    """Apply point source vs. uniform source pathloss adjustments
+    to a NIRSpec IFU 2D master background array.
+
+    Parameters
+    ----------
+    input : `~jwst.datamodels.IFUImageModel`
+        The input background data.
+
+    Returns
+    -------
+    input : `~jwst.datamodels.IFUIMAGEModel`
+        An updated (in place) version of the input with the data
+        replaced by the corrected 2D background.
+    """
+
+    log.info('Applying point source pathloss updates to IFU background')
+    input.data *= (input.pathloss_point / input.pathloss_uniform)
```

Comment:
What happens here if someone turns off the pathloss correction step?  Might want a `try` block.

Reply:
Good point. But that raises an issue I was having while trying to run my first test on this new code. I was using an old cal file that still had the old "pathloss" array, but not the new "pathloss_point" or "pathloss_uniform" arrays. I could NOT find a useful way to test for the true existence of the new arrays. Any use of `hasattr(model, 'pathloss_point')` always returns `True`, because it's defined in the schema, `model.pathloss_point.shape` returns a valid looking `(2048,2048)`, etc. About the only way I could tell whether the data were real or not was to see if the data array was filled with zeros (which for some kinds of corrections could be valid). How can you reliably test for the true existence of an attribute like this in our datamodels?!

Reply:
That is quite irritating.  @stscieisenhamer is there a trick for doing this?